Makefile

```
@@ -1291,7 +1291,7 @@ jsc-build-linux-compile-config:
 jsc-build-linux-compile-build:
 		mkdir -p $(WEBKIT_RELEASE_DIR)  && \
 		cd $(WEBKIT_RELEASE_DIR)  && \
-	CFLAGS="$(CFLAGS) -Wl,--whole-archive -ffat-lto-objects" CXXFLAGS="$(CXXFLAGS) -Wl,--whole-archive -ffat-lto-objects" -DUSE_BUN_JSC_ADDITIONS=ON \
+	CFLAGS="$(CFLAGS) -Wl,--whole-archive -ffat-lto-objects" CXXFLAGS="$(CXXFLAGS) -Wl,--whole-archive -ffat-lto-objects" \
```

This flag needs to be enabled. The bug is that `-DUSE_BUN_JSC_ADDITIONS=ON` is not inside `CXXFLAGS`
```suggestion
	CFLAGS="$(CFLAGS) -Wl,--whole-archive -ffat-lto-objects" CXXFLAGS="$(CXXFLAGS) -Wl,--whole-archive -ffat-lto-objects -DUSE_BUN_JSC_ADDITIONS=ON" \
```

---

Thanks! I noticed I'm still getting linker errors with `make jsc` on Linux.

```
ld.lld: error: undefined symbol: Bun__serializeJSValueForSubprocess
>>> referenced by javascript.zig:298 (/home/aaron/bun/src/bun.js/javascript.zig:298)
>>>               /home/aaron/bun/packages/debug-bun-linux-x64/headers.o:(Bun__Process__send)

ld.lld: error: undefined symbol: JSC__JSGlobalObject__drainMicrotasks
>>> referenced by event_loop.zig:538 (/home/aaron/bun/src/bun.js/event_loop.zig:538)
>>>               /home/aaron/bun/packages/debug-bun-linux-x64/headers.o:(src.bun.js.event_loop.EventLoop.drainMicrotasksWithGlobal)

ld.lld: error: undefined symbol: Process__emitDisconnectEvent
>>> referenced by javascript.zig:2771 (/home/aaron/bun/src/bun.js/javascript.zig:2771)
>>>               /home/aaron/bun/packages/debug-bun-linux-x64/headers.o:(src.bun.js.javascript.VirtualMachine.IPCInstance.handleIPCClose)

ld.lld: error: undefined symbol: Bun__JSValue__deserialize
>>> referenced by ipc.zig:43 (/home/aaron/bun/src/bun.js/ipc.zig:43)
>>>               /home/aaron/bun/packages/debug-bun-linux-x64/headers.o:(src.bun.js.ipc.decodeIPCMessage)

ld.lld: error: undefined symbol: Process__emitMessageEvent
>>> referenced by javascript.zig:2761 (/home/aaron/bun/src/bun.js/javascript.zig:2761)
>>>               /home/aaron/bun/packages/debug-bun-linux-x64/headers.o:(src.bun.js.javascript.VirtualMachine.IPCInstance.handleIPCMessage)

ld.lld: error: undefined symbol: ubrk_following_70
>>> referenced by StringView.cpp
>>>               StringView.cpp.o:(WTF::StringView::GraphemeClusters::Iterator::Iterator(WTF::StringView, unsigned int)) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libWTF.a
>>> referenced by StringView.cpp
>>>               StringView.cpp.o:(WTF::StringView::GraphemeClusters::Iterator::operator++()) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libWTF.a
>>> referenced by UnifiedSource-f2e18ffc-17.cpp
>>>               UnifiedSource-f2e18ffc-17.cpp.o:(JSC::IntlSegments::containing(JSC::JSGlobalObject*, JSC::JSValue)) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libJavaScriptCore.a

ld.lld: error: undefined symbol: unorm2_getNFCInstance_70
>>> referenced by StringView.cpp
>>>               StringView.cpp.o:(WTF::normalizedNFC(WTF::StringView)) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libWTF.a
>>> referenced by UnifiedSource-f2e18ffc-36.cpp
>>>               UnifiedSource-f2e18ffc-36.cpp.o:(JSC::stringProtoFuncNormalize(JSC::JSGlobalObject*, JSC::CallFrame*)) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libJavaScriptCore.a

ld.lld: error: undefined symbol: unorm2_isNormalized_70
>>> referenced by StringView.cpp
>>>               StringView.cpp.o:(WTF::normalizedNFC(WTF::StringView)) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libWTF.a
>>> referenced by UnifiedSource-f2e18ffc-36.cpp
>>>               UnifiedSource-f2e18ffc-36.cpp.o:(JSC::stringProtoFuncNormalize(JSC::JSGlobalObject*, JSC::CallFrame*)) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libJavaScriptCore.a

ld.lld: error: undefined symbol: unorm2_normalize_70
>>> referenced by StringView.cpp
>>>               StringView.cpp.o:(WTF::normalizedNFC(WTF::StringView)) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libWTF.a
>>> referenced by StringView.cpp
>>>               StringView.cpp.o:(WTF::normalizedNFC(WTF::StringView)) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libWTF.a
>>> referenced by UnifiedSource-f2e18ffc-36.cpp
>>>               UnifiedSource-f2e18ffc-36.cpp.o:(JSC::stringProtoFuncNormalize(JSC::JSGlobalObject*, JSC::CallFrame*)) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libJavaScriptCore.a
>>> referenced 1 more times

ld.lld: error: undefined symbol: u_strToLower_70
>>> referenced by StringImpl.cpp
>>>               StringImpl.cpp.o:(WTF::StringImpl::convertToLowercaseWithoutLocale()) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libWTF.a
>>> referenced by StringImpl.cpp
>>>               StringImpl.cpp.o:(WTF::StringImpl::convertToLowercaseWithoutLocale()) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libWTF.a
>>> referenced by StringImpl.cpp
>>>               StringImpl.cpp.o:(WTF::StringImpl::convertToLowercaseWithLocale(WTF::AtomString const&)) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libWTF.a
>>> referenced 3 more times

ld.lld: error: undefined symbol: u_tolower_70
>>> referenced by StringImpl.cpp
>>>               StringImpl.cpp.o:(WTF::StringImpl::convertToLowercaseWithoutLocaleStartingAtFailingIndex8Bit(unsigned int)) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libWTF.a
>>> referenced by UnifiedSource-6cbc989f-1.cpp
>>>               UnifiedSource-6cbc989f-1.cpp.o:(JSC::Yarr::ByteCompiler::atomPatternCharacter(int, JSC::Yarr::MatchDirection, unsigned int, unsigned int, WTF::Checked<unsigned int, WTF::CrashOnOverflow>, JSC::Yarr::QuantifierType)) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libJavaScriptCore.a

ld.lld: error: undefined symbol: u_toupper_70
>>> referenced by StringImpl.cpp
>>>               StringImpl.cpp.o:(WTF::StringImpl::convertToUppercaseWithoutLocale()) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libWTF.a
>>> referenced by StringImpl.cpp
>>>               StringImpl.cpp.o:(WTF::StringImpl::convertToUppercaseWithoutLocale()) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libWTF.a
>>> referenced by StringImpl.cpp
>>>               StringImpl.cpp.o:(WTF::StringImpl::convertToUppercaseWithoutLocale()) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libWTF.a
>>> referenced 1 more times

ld.lld: error: undefined symbol: u_strToUpper_70
>>> referenced by StringImpl.cpp
>>>               StringImpl.cpp.o:(WTF::StringImpl::convertToUppercaseWithoutLocale()) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libWTF.a
>>> referenced by StringImpl.cpp
>>>               StringImpl.cpp.o:(WTF::StringImpl::convertToUppercaseWithoutLocale()) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libWTF.a
>>> referenced by StringImpl.cpp
>>>               StringImpl.cpp.o:(WTF::StringImpl::convertToUppercaseWithLocale(WTF::AtomString const&)) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libWTF.a
>>> referenced 3 more times

ld.lld: error: undefined symbol: u_foldCase_70
>>> referenced by StringImpl.cpp
>>>               StringImpl.cpp.o:(WTF::StringImpl::foldCase()) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libWTF.a

ld.lld: error: undefined symbol: u_strFoldCase_70
>>> referenced by StringImpl.cpp
>>>               StringImpl.cpp.o:(WTF::StringImpl::foldCase()) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libWTF.a
>>> referenced by StringImpl.cpp
>>>               StringImpl.cpp.o:(WTF::StringImpl::foldCase()) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libWTF.a

ld.lld: error: undefined symbol: u_charDirection_70
>>> referenced by StringImpl.cpp
>>>               StringImpl.cpp.o:(WTF::StringImpl::defaultWritingDirection()) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libWTF.a
>>> referenced by StringImpl.cpp
>>>               StringImpl.cpp.o:(WTF::StringImpl::defaultWritingDirection()) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libWTF.a
>>> referenced by UnifiedSource-f2e18ffc-10.cpp
>>>               UnifiedSource-f2e18ffc-10.cpp.o:(JSC::invalidParameterInSourceAppender(WTF::String const&, WTF::StringView, JSC::RuntimeType, JSC::ErrorInstance::SourceTextWhereErrorOccurred)) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libJavaScriptCore.a
>>> referenced 3 more times

ld.lld: error: undefined symbol: ubrk_open_70
>>> referenced by TextBreakIterator.cpp
>>>               TextBreakIterator.cpp.o:(WTF::wordBreakIterator(WTF::StringView)) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libWTF.a
>>> referenced by TextBreakIterator.cpp
>>>               TextBreakIterator.cpp.o:(WTF::sentenceBreakIterator(WTF::StringView)) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libWTF.a
>>> referenced by TextBreakIterator.cpp
>>>               TextBreakIterator.cpp.o:(WTF::NonSharedCharacterBreakIterator::NonSharedCharacterBreakIterator(WTF::StringView)) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libWTF.a
>>> referenced 3 more times

ld.lld: error: undefined symbol: ubrk_setUText_70
>>> referenced by TextBreakIterator.cpp
>>>               TextBreakIterator.cpp.o:(WTF::setTextForIterator(UBreakIterator&, WTF::StringView)) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libWTF.a
>>> referenced by TextBreakIterator.cpp
>>>               TextBreakIterator.cpp.o:(WTF::TextBreakIteratorICU::TextBreakIteratorICU(WTF::StringView, char16_t const*, unsigned int, std::variant<WTF::TextBreakIteratorICU::LineMode, WTF::TextBreakIteratorICU::CharacterMode>, WTF::AtomString const&)) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libWTF.a

ld.lld: error: undefined symbol: utext_close_70
>>> referenced by TextBreakIterator.cpp
>>>               TextBreakIterator.cpp.o:(WTF::setTextForIterator(UBreakIterator&, WTF::StringView)) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libWTF.a
>>> referenced by TextBreakIterator.cpp
>>>               TextBreakIterator.cpp.o:(WTF::TextBreakIteratorICU::TextBreakIteratorICU(WTF::StringView, char16_t const*, unsigned int, std::variant<WTF::TextBreakIteratorICU::LineMode, WTF::TextBreakIteratorICU::CharacterMode>, WTF::AtomString const&)) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libWTF.a

ld.lld: error: undefined symbol: ubrk_setText_70
>>> referenced by TextBreakIterator.cpp
>>>               TextBreakIterator.cpp.o:(WTF::setTextForIterator(UBreakIterator&, WTF::StringView)) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libWTF.a
>>> referenced by UnifiedSource-f2e18ffc-17.cpp
>>>               UnifiedSource-f2e18ffc-17.cpp.o:(JSC::IntlSegmenter::segment(JSC::JSGlobalObject*, JSC::JSValue) const) in archive /home/aaron/bun/src/bun.js/WebKit/WebKitBuild/Release/lib/libJavaScriptCore.a
```

---

Are you sure it needs to be in the CXXFLAGS? It seems to be a CMAKE flag, but these aren't needed for cmake build?